app/controllers/AlertController.java

```diff
@@ -61,6 +72,89 @@ public AlertController(final Configuration configuration, final AlertRepository
     }
 
     /**
+     * Adds an alert in the alert repository.
+     *
+     * @return Created if the alert was created successfully, an error code otherwise.
+     */
+    public Result add() {
+        Alert alert;
+        try {
+            final models.view.Alert viewAlert = buildViewAlert(request().body());
+            alert = convertToInternalAlert(viewAlert);
+        } catch (final IOException | ConstraintsViolatedException | IllegalArgumentException e) {
+            LOGGER.error()
+                    .setMessage("Failed to build an alert.")
+                    .setThrowable(e)
+                    .log();
+            return badRequest("Invalid request body.");
+        }
+
+        if (_alertRepository.get(alert.getId()).isPresent()) {
```

Comment:
This is unnecessary, the model being posted will always contain the id and the code should perform an upsert. The user's intention is either to create or update. According to @matthayter and @BrandonArp this was what was discussed as the standard for the CRUD REST controllers. So to recap:

 - PUT for add/update
 - Entity contains uuid
 - Optional uuid in the post uri -- @matthayter or @BrandonArp confirm

If we need stronger semantics on update (to prevent unintentional overwrites) we will need to version the widgets. In the case of @matthayter -'s evergreen project that is not necessary for the first version. For your alerts/expressions project that would probably be a good thing. One of the next features will be change history for alerts and expressions so that will require some versioning work anyway (we can leave the change conflict work until then).